screens/dashboard/components/upcoming.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import moment from 'moment';
+
+// import global actions
+import * as actions from '../actions';
+
+// import global styles
+// @TODO: move the global styles from this screen into ../../styles/common 
+import commonStyles from '../../../styles/common';
+const styles = StyleSheet.create(commonStyles);
+
+type Props = {
+  actions: Object,
+  profile: Object,
+  navigation: Object
+};
+
+
+export class Upcoming extends Component<Props> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+    this.icons = {
+      'open': 'angle-down',
+      'close': 'angle-up'
+    };
+  }
+
+
+  showUpcoming(firstOrRemaining) {
+
+    let upcomingArray = ((this.props.profile || {}).upcomingArray || {});
+
+    let allUpcomingEvents = [];
+    for (i = 0; i < upcomingArray.length; i++) {
+
+      let title = ((((this.props.profile || {}).upcomingArray || {})[i] || {}).title);
+      let location = ((((this.props.profile || {}).upcomingArray || {})[i] || {}).location);
+      let date = ((((this.props.profile || {}).upcomingArray || {})[i] || {}).date);
+      let momentDate = moment(new Date(((date || {}).seconds) * 1000));
+      let localDate = moment(momentDate.toISOString()).toString();
+      let formattedDate = moment(localDate).format('ddd M/D/YY h:mma');
+      let dayNumber = momentDate.toNow(true).split(' ')[0];
+      let dayWord = momentDate.toNow(true).split(' ')[1];
+
+      if (title && moment(Date.now()) < momentDate) {
+
+        if (dayNumber === "a" || dayNumber === "an") {
```

Comment:
these nested if statements could be better expressed using a switch,  or even better, create a function that calculates dayNumber and use that to assign the variable.

Reply:
```        let dayNumber = momentDate.toNow(true).split(' ')[0];
        let dayWord = momentDate.toNow(true).split(' ')[1];

        switch (true) {
          case dayNumber.includes("a") && dayWord === "few":
            dayNumber = 1;
            dayWord = "minute";
            break;
          case dayNumber.includes("a"):
            dayNumber = 1;
            break;
          case dayWord === "days" && dayNumber > 6 && dayNumber < 14:
            dayNumber = 1;
            dayWord = "week";
            break;
          case dayWord === "days" && dayNumber > 13 && dayNumber < 29:
            dayNumber = Math.floor(dayNumber / 7);
            dayWord = "weeks";
            break;
        }
```